### PR TITLE
[nT2] DQM for the TOTEM nT2 detector

### PR DESCRIPTION
**Pull request by rozchlastywacz**
#### PR description:

This PR introduces Data Quality Monitor for T2 telescope.
This code works on top of recently merged PR: https://github.com/cms-sw/cmssw/pull/38986 

#### PR validation:
Code compiles and the `DQM/CTPPS/test/totemt2_dqm_test_fromdigi_cfg.py` test produces DQM root file with plots as expected. Because detector is not yet integrated, test is run on emulated data.`/store/group/dpg_ctpps/comm_ctpps/TotemT2/RecoTest/emulated_digi_test.root`.
Sample plot from DQM output file:
image

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:
This PR is NOT a backport, but it is intented to be integrated into CMSSW_12_5_X.

@forthommel @grzanka @ChrisMisan for your attention.

### CODE REVIEW

**Comment by ChrisMisan** [comment]
type ctpps, new-feature

**Comment by ChrisMisan** [comment]
please test

**Comment by ChrisMisan** [comment]
@cms-sw/dqm-l2 please consider signing this, it is not changing any of the existing code.

**Comment by forthommel** [comment]
@civanch the two commits in this PR have been stashed to reflect the history of development for this package.

**Comment by forthommel** [comment]
With https://github.com/cms-sw/cmssw/pull/39159#event-7250429085 I rebased this branch to the recently-merged #39140 replacing the digis input collection from an `edm::DetSetVector<TotemT2Digi>` to a `edmNew::DetSetVector<TotemT2Digi>`.

**Comment by emanueleusai** [comment]
please test

**Comment by forthommel** [comment]
Can we still aim at having this reviewed before the weekend? :)

**Comment by emanueleusai** [comment]
+1

File: DQM/CTPPS/plugins/TotemT2DQMSource.cc [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+  std::string title, path;
+  TotemT2DetId(id).planeName(title, TotemT2DetId::nFull);
+  TotemT2DetId(id).planeName(path, TotemT2DetId::nPath);
+  ibooker.setCurrentFolder(path);
+
+  digisMultiplicity = ibooker.book2DD("digis multiplicity",
+                                      title + " digis multiplicity;arbitrary unit;arbitrary unit",
+                                      nbinsx,
+                                      0.,
+                                      nbinsx,
```
Comment (perrotta) [inline DQM/CTPPS/plugins/TotemT2DQMSource.cc:134]: This parameter should be a `double`
Then, for some reasonable histogram that reports integers in nbinsx bins, I imagine that it should better go from `-0.5` to `double(nbinsx)-0.5` (same for `nbinsy`)

File: DQM/CTPPS/plugins/TotemT2DQMSource.cc [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+    mask[4 * pl + channel] = true;
+  const TotemT2DetId secId(detid.armId());
+  // check how many masked channels were hit
+  unsigned int triggeredChannelsNumber = (mask & sectorPlots_[secId].hitTilesArray).count();
+
+  return triggeredChannelsNumber >= SectorPlots::MINIMAL_TRIGGER;
+}
+
+void TotemT2DQMSource::bookErrorFlagsHistogram(DQMStore::IBooker& ibooker) {
+  HPTDCErrorFlags_2D_ = ibooker.book2D("HPTDC Errors", " HPTDC Errors?", 8, -0.5, 8.5, 2, -0.5, 1.5);
```
Comment (perrotta) [inline DQM/CTPPS/plugins/TotemT2DQMSource.cc:256]: Either
```suggestion
  HPTDCErrorFlags_2D_ = ibooker.book2D("HPTDC Errors", " HPTDC Errors?", 8, -0.5, 7.5, 2, -0.5, 1.5);
```
or
```suggestion
  HPTDCErrorFlags_2D_ = ibooker.book2D("HPTDC Errors", " HPTDC Errors?", 9, -0.5, 8.5, 2, -0.5, 1.5);
```

File: DQM/CTPPS/plugins/TotemT2DQMSource.cc [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+  activePlanes = ibooker.book1D("active planes", title + " which planes are active;plane number", 8, -0.5, 7.5);
+
+  activePlanesCount = ibooker.book1D(
+      "number of active planes", title + " how many planes are active;number of active planes", 9, -0.5, 8.5);
+
+  triggerEmulator = ibooker.book2DD("trigger emulator",
+                                    title + " trigger emulator;arbitrary unit;arbitrary unit",
+                                    nbinsx,
+                                    0.,
+                                    nbinsx,
```
Comment (perrotta) [inline DQM/CTPPS/plugins/TotemT2DQMSource.cc:108]: This parameter should be a `double`
Then, for some reasonable histogram that reports integers in nbinsx bins, I imagine that it should better go from `-0.5` to `double(nbinsx)-0.5` (same for `nbinsy`)

File: DQM/CTPPS/plugins/TotemT2DQMSource.cc [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+  digisMultiplicity = ibooker.book2DD("digis multiplicity",
+                                      title + " digis multiplicity;arbitrary unit;arbitrary unit",
+                                      nbinsx,
+                                      0.,
+                                      nbinsx,
+                                      nbinsy,
+                                      0.,
+                                      nbinsy);
+  rechitMultiplicity = ibooker.book2DD(
+      "rechits multiplicity", title + " rechits multiplicity;x;y", nbinsx, 0., nbinsx, nbinsy, 0., nbinsy);
```
Comment (perrotta) [inline DQM/CTPPS/plugins/TotemT2DQMSource.cc:139]: Same here

**Comment by civanch** [comment]
please test

**Comment by emanueleusai** [comment]
+1

- re-sign

**Comment by perrotta** [comment]
+1

**Pull request merged**